scripts/linking.py

```diff
@@ -482,7 +535,7 @@ def main(medmentions_path: str,
         thresholds = [float(x) for x in thresholds.split(",")]
 
     # only evaluate on the dev examples for now because we don't have a trained model
-    eval_candidate_generation(dev_examples, umls_concept_dict_by_id, candidate_generator, k_list, thresholds, use_gold_mentions, spacy_model)
+    eval_candidate_generation_and_linking(dev_examples, umls_concept_dict_by_id, candidate_generator, linking_classifier, k_list, thresholds, use_gold_mentions, spacy_model)
```

Comment:
As above, can you make the linking classifier separate and optional.

Reply:
done

Reply:
I made it optional and moved almost all its code in its own class, but it is still being called from this function. Moving it outside will result into a lot of duplicate code.